tez-runtime-internals/src/main/java/org/apache/tez/runtime/task/TezChild.java

```diff
@@ -503,6 +506,33 @@ public static TezChild newTezChild(Configuration conf, String host, int port, St
         hadoopShim);
   }
 
+  private static void eagerInitFileSystemPaths(Configuration conf) {
```

Comment:
It will be good to measure the time spent on FS init (even for cloudstores) and share the details, before trying out this patch.

Reason: This get inited in TezChild, but for running container, it closes the FileSystem explicitly via "FileSystem.closeAllForUGI(childUGI);". Refer "public ContainerExecutionResult run()" method.

Even if this gets early inited, it will not have major impact in container reuse scenario. It will be good to measure and find out the  timing spent in FS init.